opentelemetry-semantic-conventions/src/opentelemetry/semconv/_incubating/attributes/messaging_attributes.py

```diff
@@ -21,11 +21,16 @@
 Note: Instrumentations SHOULD NOT set `messaging.batch.message_count` on spans that operate with a single message. When a messaging client library supports both batch and single-message API for the same operation, instrumentations SHOULD use `messaging.batch.message_count` for batching APIs and SHOULD NOT use it for single-message APIs.
 """
 
-MESSAGING_CLIENT_ID = "messaging.client_id"
+MESSAGING_CLIENT_ID = "messaging.client.id"
 """
 A unique identifier for the client that consumes or produces a message.
 """
 
+MESSAGING_CLIENT_ID = "messaging.client_id"
```

Comment:
how come it does not raise any errors in CI ? (there is a duplication now) - https://github.com/open-telemetry/semantic-conventions/issues/1031


I can join python SIG tomorrow to discuss the options:
1. we can drop `MESSAGING_CLIENT_ID = "messaging.client_id"`, the only downside that someone's telemetry might magically change and if they relied on `messaging.client_id` it will be replaced by `messaging.client.id`. It's incubating, experimental thing so this could be tolerable.
2. we can generate two constants, e.g.:
  - `MESSAGING_CLIENT_ID2 = "messaging.client.id"` (once stabilizes it'd appear as `MESSAGING_CLIENT_ID`  in a stable part.
  - `MESSAGING_CLIENT_ID = "messaging.client_id"`

happy to help with jinja, but want to hear maintainer's opinions

Reply:
Nobody is using MESSAGING_CLIENT_ID in contrib repo

Reply:
user apps might, but it'd still be a behavior-only breaking change in internal API of experimental artifact :)

Reply:
https://github.com/xrmx/opentelemetry-python/pull/1 shows how to drop old attribute.

Reply:
> how come it does not raise any errors in CI ? (there is a duplication now) - https://github.com/open-telemetry/semantic-conventions/issues/1031

Apparently there is no pylint rule that checks that constant is not reassigned 😱

I can update jinja to detect conflicts and fail code generation if it detects any as a follow up.

Reply:
> user apps might, but it'd still be a behavior-only breaking change in internal API of experimental artifact :)

Since our components in OpenTelemetry does not reference it, and semconv itself is still experimental, I am okay with just simply making the breaking change. In the future when semconv becomes 1.X, we must use different strategies for deprecation (as outlined here https://github.com/open-telemetry/opentelemetry-python/pull/3586#issuecomment-1973978521)

Reply:
@lmolkova please open a pr with the scripts changes against upstream. I'll rebase this on top of yours

Reply:
here you go - https://github.com/open-telemetry/opentelemetry-python/pull/3966

Reply:
Rebases and added the duplicated attribute in EXCLUDED_ATTRIBUTES